module/segment/src/Infrastructure/Service/SegmentProductService.php

```diff
@@ -48,21 +48,20 @@ public function add(SegmentId $segmentId, ProductId $productId): void
      */
     public function mark(SegmentId $segmentId, ProductId $productId): void
     {
-        $sql = 'INSERT INTO '.self::TABLE.' (segment_id, product_id, available, calculated_at)
-            VALUES (:segmentId, :productId, true, :calculatedAt)
-            ON CONFLICT ON CONSTRAINT segment_product_pkey
-                DO UPDATE SET available = true
-        ';
-        $this->connection->executeQuery(
-            $sql,
+        $this->connection->update(
```

Comment:
Is it intended to "only" update existing entries?

Reply:
yes, problem was in other place where wy dont add entry earlier